Mathlib/Topology/Order/LawsonTopology.lean

```diff
@@ -0,0 +1,252 @@
+      (upperClosure t : Set α)ᶜ ∩ u = s }
+
+protected theorem isTopologicalBasis : TopologicalSpace.IsTopologicalBasis (lawson_basis α) := by
+  have lawson_basis_image2 : lawson_basis α =
+      (image2 (fun x x_1 ↦ ⇑WithLower.toLower ⁻¹' x ∩ ⇑WithScott.toScott ⁻¹' x_1)
+        (IsLower.lowerBasis (WithLower α)) {U | @IsOpen (WithScott α) _ U}) := by
+    rw [lawson_basis, image2, IsLower.lowerBasis]
+    aesop
+  rw [lawson_basis_image2]
+  convert IsTopologicalBasis.inf_induced IsLower.isTopologicalBasis
+    (TopologicalSpace.isTopologicalBasis_opens (α := WithScott α))
+    WithLower.toLower WithScott.toScott
+  erw [topology_eq α]
+  rw [lawson]
+  apply (congrArg₂ Inf.inf _) _
+  letI _ := lower α; exact @IsLower.withLowerHomeomorph α ‹_› (lower α) ⟨rfl⟩ |>.inducing.induced
+  letI _ := scott α; exact @IsScott.withScottHomeomorph α _ (scott α) ⟨rfl⟩ |>.inducing.induced
+
+end Preorder
+end IsLawson
+
+/--
+Type synonym for a preorder equipped with the Lawson topology
+-/
+def WithLawson (α : Type*) := α
+
+namespace WithLawson
+
+/-- `toLawson` is the identity function to the `WithLawson` of a type.  -/
+@[match_pattern] def toLawson : α ≃ WithLawson α := Equiv.refl _
+
+/-- `ofLawson` is the identity function from the `WithLawson` of a type.  -/
+@[match_pattern] def ofLawson : WithLawson α ≃ α := Equiv.refl _
+
+@[simp] lemma to_Lawson_symm_eq : (@toLawson α).symm = ofLawson := rfl
+@[simp] lemma of_Lawson_symm_eq : (@ofLawson α).symm = toLawson := rfl
+@[simp] lemma toLawson_ofLawson (a : WithLawson α) : toLawson (ofLawson a) = a := rfl
+@[simp] lemma ofLawson_toLawson (a : α) : ofLawson (toLawson a) = a := rfl
+
+@[simp, nolint simpNF]
+lemma toLawson_inj {a b : α} : toLawson a = toLawson b ↔ a = b := Iff.rfl
+
+@[simp, nolint simpNF]
+lemma ofLawson_inj {a b : WithLawson α} : ofLawson a = ofLawson b ↔ a = b :=
+Iff.rfl
+
+/-- A recursor for `WithLawson`. Use as `induction x using WithLawson.rec`. -/
+protected def rec {β : WithLawson α → Sort _}
+    (h : ∀ a, β (toLawson a)) : ∀ a, β a := fun a => h (ofLawson a)
+
+instance [Nonempty α] : Nonempty (WithLawson α) := ‹Nonempty α›
+instance [Inhabited α] : Inhabited (WithLawson α) := ‹Inhabited α›
+
+variable [Preorder α]
+
+instance : Preorder (WithLawson α) := ‹Preorder α›
+instance : TopologicalSpace (WithLawson α) := lawson α
+instance : IsLawson (WithLawson α) := ⟨rfl⟩
+
+/-- If `α` is equipped with the Lawson topology, then it is homeomorphic to `WithLawson α`.
+-/
+def withLawsonTopologyHomeomorph [TopologicalSpace α] [IsLawson α]  : WithLawson α ≃ₜ α :=
+  WithLawson.ofLawson.toHomeomorphOfInducing ⟨by erw [IsLawson.topology_eq α, induced_id]; rfl⟩
+
+theorem isOpen_preimage_ofLawson (S : Set α) :
+    IsOpen (Topology.WithLawson.ofLawson ⁻¹' S) ↔
+      (lawson α).IsOpen S :=
+  Iff.rfl
+
+theorem isClosed_preimage_ofLawson (S : Set α) :
+    IsClosed (Topology.WithLawson.ofLawson ⁻¹' S) ↔
+      @IsClosed (WithLawson α) _ S :=
+  Iff.rfl
```

Comment:
Possibly - I still don't have a good grasp of how to think about multiple topologies on the same space in Mathlib, so I'm happy to go with this if it's the better way of doing it.

Reply:
`@IsClosed (WithLawson α) _ S` is not well-typed. The way to fix it would be to replace `S` with `ofLawson ⁻¹' S`, in which case the statement becomes `IsClosed (ofLawson ⁻¹' S) ↔ IsClosed (ofLawson ⁻¹' S)`, namely a tautology.

Reply:
I only use `IsLawson.isClosed_preimage_ofLawson` in `singleton_isClosed`. `WithLawson.isClosed_preimage_ofLawson` doesn't seem to work there.

Reply:
This works:
```
lemma singleton_isClosed (a : α) : IsClosed ({a} : Set α) := by
  simp only [IsLawson.topology_eq]
  rw [← (Set.OrdConnected.upperClosure_inter_lowerClosure ordConnected_singleton),
    ← WithLawson.isClosed_preimage_ofLawson]
  apply IsClosed.inter
    (lawsonClosed_of_lowerClosed _ (IsLower.isClosed_upperClosure (finite_singleton a)))
  rw [ lowerClosure_singleton, LowerSet.coe_Iic, ← WithLawson.isClosed_preimage_ofLawson]
  apply lawsonClosed_of_scottClosed
  exact IsScott.isClosed_Iic
```
So we could get rid of `IsLawson.isClosed_preimage_ofLawson`?

Reply:
Sure yeah